src/network/postgres/postgres_protocol_interpreter.cpp

```diff
@@ -112,7 +113,7 @@ bool PostgresProtocolInterpreter::TryReadPacketHeader(const std::shared_ptr<Read
   // Extend the buffer as needed
   if (curr_input_packet_.len_ > in->Capacity()) {
     // Allocate a larger buffer and copy bytes off from the I/O layer's buffer
-    curr_input_packet_.buf_ = std::make_shared<ReadBuffer>(curr_input_packet_.len_);
+    curr_input_packet_.buf_ = std::make_unique<ReadBuffer>(curr_input_packet_.len_);
     NETWORK_LOG_TRACE("Extended Buffer size required for packet of size {0}", curr_input_packet_.len_);
     curr_input_packet_.extended_ = true;
   } else {
```

Comment:
So in the line below this one, that's when we currently force the shared_ptr usage. Can we get around this?